schema_based_on_diagram.sql

```diff
@@ -0,0 +1,58 @@
+--create patients table
+CREATE TABLE patients(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	name CHAR(255),
+	date_of_birth DATE
+);
+
+-- create medical_hitories
+CREATE TABLE medical_histories(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	admitted_at TIME,
+	patient_id INT,
+	status CHAR(255)
+);
+
+-- create invoices table
+
+CREATE TABLE invoices(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	total_amount DECIMAL,
+	generated_at TIME,
+	payed_at TIME,
+	medical_history_id INT
+);
+
+--create invoice_items
+
+CREATE TABLE invoice_items(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	unit_price DECIMAL,
+	quantity INT,
+	total_price DECIMAL,
+	invoice_id INT,
+	treatment_id INT
+);
+
+-- create treatments table
+CREATE TABLE treatments(
+	id INT PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+	type CHAR(255),
+	name CHAR(255)
+);
+
```

Comment:
- [ ] Last but not least, there is a requirement that we are yet to fulfill. Which is this 👇🏼:

  > - Remember to add the FK indexes.
  
  Please look at the schema and index all the foreign keys. An example would be something like this 👇🏼:
  
  ```SQL
  CREATE INDEX ON medical_histories (patient_id);
  ```
  
  Now go through the schema and find other Foreign Keys and index them as I have done with the `patient_id` FK up there ☝🏼.

Reply:
Thanks again for the meticulous feedback :+1: